### PR TITLE
Add native PTZ element

### LINKED ISSUE #798: Native PTZ controls
**Is your feature request related to a problem? Please describe.**
Apart from https://github.com/dermotduffy/frigate-hass-card/issues/797, I think some "native" PTZ controls would be good for the following reasons:

1. No need to calculate the pixels for each arrow. This is boring, but the problematic part is that they get displayed differently based on the device's DPI. It would be better if all the buttons were grouped and aligned centrally or something like that (instead of based on pixels only).
2. Difficult to apply fancy styles for each button, or something like the one in the WebRTC card which looks like a wheel (single image but clickable in 4 different areas).
3. Reduce a lot of YAML code duplication. Let's say you have 3 cards with such PTZ controls, if you want to keep them all the same, it would start to be a pain.

**Describe the solution you'd like**
This should be a rather popular use case. Having native support would simplify the people's setup, and would unlock the ability to make it fancier. I believe if the code from WebRTC could be borrowed, it should, because I think it works really well.

**Describe alternatives you've considered**
To use custom elements and customize them by myself. 

**Additional context**
If you disagree, I totally respect. The current approach isn't bad at all.

### PR DESCRIPTION
**Pull request by dermotduffy**
*  Closes #798 

```yaml
elements:
  - type: custom:frigate-card-conditional
    conditions:
      mediaLoaded: true
      view:
        - live
    elements:
      - type: custom:frigate-card-ptz
        orientation: horizontal
        service: homeassistant.toggle
        style:
          transform: none
          right: 20px
          top: 180px
        data_left:
          entity_id: light.office_main_lights
```

### CODE REVIEW

**Comment by dermotduffy** [comment]
@felipecrs Give this a shot? I'm not sure if this should be a custom element, or 'builtin' but went with element for now. Let me know what you think. You can set `orientation` to `horizontal` or `vertical`.

**Comment by felipecrs** [comment]
I'll surely try ASAP.

**Comment by felipecrs** [comment]
It looks great, but I think you should allow the user to select some preset position instead of asking them to make it pixel-based (which is one of the problems I reported: pixel-based locations do not work well when in different DPIs).

https://user-images.githubusercontent.com/29582865/185723047-9668ec99-3149-4357-b38f-da121495e283.mp4

How about a new attribute like `position`, which can be: `left`, `center`, `right` and another attribute called `alignment` which could be `top`, `center`, `bottom`. So that I could set the custom element to be positioned at bottom-right.

And I think it's a good decision keeping it as a custom element.

**Comment by dermotduffy** [comment]
> (which is one of the problems I reported: pixel-based locations do not work well when in different DPIs).

I'm not following why you think this will be different in code, unless you want active measurement of some kind. i.e. why can't you just put the positions in %s so that they scale with the card viewport, e.g. have you tried:

```
      - type: custom:frigate-card-ptz
        orientation: horizontal
        service: homeassistant.toggle
        style:
          transform: none
          right: 5%
          top: 50%
[...]
```

PS: If you have a specific test scenario / config let me know.

> How about a new attribute like position, which can be: left, center, right and another attribute called alignment which could be top, center, bottom. So that I could set the custom element to be positioned at bottom-right.

I definitely could do this, although it felt like a fairly personal choice that presets may just get in the way off. Also I could imagine you need different values depending on where you have the menu, whether you have thumbnails on/off, whether you have the next/previous controls, etc -- much easier to just the user put it wherever they want.

**Comment by felipecrs** [comment]
> I'm not following why you think this will be different in code, unless you want active measurement of some kind. i.e. why can't you just put the positions in %s so that they scale with the card viewport, e.g. have you tried:
> 
> ```
>       - type: custom:frigate-card-ptz
>         orientation: horizontal
>         service: homeassistant.toggle
>         style:
>           transform: none
>           right: 5%
>           top: 50%
> [...]
> ```

It works like a charm. I would only suggest to add this to the `README`, as I think it should work in more cases than what is there currently:

```
style:
  transform: none
  right: 20px
  top: 180px
```



> I definitely could do this, although it felt like a fairly personal choice that presets may just get in the way of. Also I could imagine you need different values depending on where you have the menu, whether you have thumbnails on/off, whether you have the next/previous controls, etc -- much easier to just the user put it wherever they want.

Sorry, I didn't realize I could do what you suggested earlier, and that covers all my needs.

I would say this PR is ready, and I like it a lot!

File: README.md [hunk 1]
```diff
@@ -1663,6 +1677,31 @@ elements:
           state: on
           state_not: off
       mediaLoaded: true
+    # Full form PTZ actions (only left button shown).
+  - type: custom:frigate-card-ptz
+    orientation: vertical
+    style:
+      transform: none
+      right: 20px
+      top: 180px
```
Comment (felipecrs) [inline README.md:1686]:
```suggestion
    style:
      transform: none
      right: 5%
      top: 50%
```

Comment (felipecrs) [reply README.md:1686]: For some reason, 45% is what actually makes it properly centered vertically. Not sure whether this is a bug or not.

**Comment by felipecrs** [comment]
Another feedback: even when using `mediaLoaded: true`, the PTZ buttons shows on top of the reconnecting message. I also noticed it being displayed on top of the Frigate splash image (the bird) when it quickly displayed.

Screenshot_20220830-170832_Home Assistant

File: README.md [hunk 2]
```diff
@@ -2147,7 +2186,7 @@ You can restrict icons to only show for certain [views](#views) using a
 `custom:frigate-card-conditional` element (e.g. PTZ controls)
 
 <details>
-  <summary>Expand: View-based conditions (e.g. PTZ controls)</summary>
+  <summary>Expand: View-based conditions (e.g. custom PTZ controls)</summary>
 
 This example shows PTZ icons that call a PTZ service, but only in the `live` view.
```
Comment (felipecrs) [inline README.md:2191]: I think this section could be deleted entirely after introducing the new ptz element.

**Comment by felipecrs** [comment]
> Another feedback: even when using `mediaLoaded: true`, the PTZ buttons shows on top of the reconnecting message. I also noticed it being displayed on top of the Frigate splash image (the bird) when it quickly displayed.

Nailing down the issue, it seems to only happen after it loaded once:


https://user-images.githubusercontent.com/29582865/187540515-e96f6621-b6d2-448d-a7b5-853ab52eff90.mp4

File: README.md (hunk 2 above)
Comment (dermotduffy) [reply README.md:2191]: I'll just change the example to non-PTZ (the intent of the example is more about conditions).

**Comment by dermotduffy** [comment]
@felipecrs I believe I have fixed those issues. I still found I can very occasionally get it to happen (the PTZ rendering on top of the loading spinner) but that is because JSMPEG appears to be actually reporting the canvas as having been loaded when it is not. I haven't tracked that down further, as I think this is good enough for now.

**Comment by felipecrs** [comment]
Awesome! :D

**Pull request merged**